### DISCUSSION TITLE
Graph algorithms

### DISCUSSION THREAD

**Discussion by ghost** [ArcadeData/arcadedb#441] [General]
What would be the best way to support graph algorithms such as shortest path, centrality, Louvian community detection, custom pregel algos, or Node2Vec on Arcade DB, similarly to memgraph's mage?

**Comment by lvca** [comment]
You can write a new function per algorithm. Look at Dijkstra:

https://github.com/ArcadeData/arcadedb/blob/39eacc45d8e65ac365c1698bb739831660e15161/engine/src/main/java/com/arcadedb/query/sql/function/graph/SQLFunctionDijkstra.java#L26-L25

And this is one of the test cases:

https://github.com/ArcadeData/arcadedb/blob/39eacc45d8e65ac365c1698bb739831660e15161/engine/src/test/java/com/arcadedb/query/sql/functions/graph/SQLFunctionDijkstraTest.java#L57

To register a new function in the SQL engine, you can just call:

```
    register(SQLFunctionDijkstra.NAME, SQLFunctionDijkstra.class);
```

See DefaultSQLFunctionFactory.java.

**Comment by lvca** [comment]
It would be wonderful to see more algorithms coming up for ArcadeDB!
